felix/statusrep/status_file_reporter.go

```diff
@@ -55,6 +58,8 @@ type EndpointStatusFileReporter struct {
 	reapplyInterval time.Duration
 
 	hostname string
+
+	*sync.Mutex
```

Comment:
```suggestion
	lock sync.Mutex
```
The zero value of `sync.Mutex` is 100% valid so you don't need to have a pointer and use `new`  You can just leave it initialised and start using it.

I generally prefer not to embed the mutex in the object (and thuse expose public Lock/Unlock methods).  The mutex is an implementation detail; you should be able to swap it for channels, atomics or something else as needed down the line without worrying that someone external might call Lock/Unlock.